### ISSUE TITLE
Problems using Shadow component since release 2.11.0

### ISSUE THREAD

**Issue by Laerso-nos** [rdkcentral/Lightning-UI-Components#383]
Hello there!!!

Guys, I have been facing an error using the Shadow component. 
Previously we had @lightningjs/ui-components in release 2.2.4, however when we updated to latest version we started to get an error in the colsole and our UI navigation breaks.
The error in the console is this: 
  `TypeError: undefined is not an object (evaluating 'v.toString') lightning-inspect.js:633`
  ```
   Object.defineProperty(ElementCore.prototype, '_colorUl', {
        get: function() {
            return this.__colorUl;
        },
        set: function(v) {
            if (this.__colorUl !== v) {
                val(this, 'colorUl', v.toString(16), "ffffffff"); //Line where error occurs
                this.__colorUl = v;
                checkColors(this);
            }
        }
    });
  ```
  
  After investigation we found out that until version 2.10.0 It works fine, but from 2.11.0 we start getting this error, we also tried to update the code and use different values from what we had, but nothing seems to work and the error is still there. This is our code using the Shadow component:  
  ```
  import { Shadow } from '@lightningjs/ui-components'
  //ofc it has the whole lightning structure around it, only pasted here the shadow part
  
  Dropshadow: {
          w: 400,
          h: 300,
          type: Shadow,
          style: {
            alpha: 0.6,
            blur: 60,
            color: 0xFF000000,
            offsetY: 44,
            radius: 8,
            spread: 2,
          },
        },
  ```
  As the error seems related to the color of the component we tried '#000000' and Colors('black').get().
  We also analyzed the changes from releases 2.10.0 to 2.11.0 and we saw nothing related to the Shadow component, also none of the changes made us think "could be this or that".
  
  Is there anything we might be missing? 
  A new approach on the Shadow usage/implementation is necessary on our side?
  
  Thank you for your support.
  Best regards

**Comment by ricardo-a-alves-alb** [comment]
@erautenberg maybe this is related to the 2 issues we discussed on Discord.

@Laerso-nos, can you try the following?:

Change this line from:

https://github.com/rdkcentral/Lightning-UI-Components/blob/c6a4caac3ff0632b8369ff9027eac30dc9a78bd4/packages/%40lightningjs/ui-components/src/mixins/withThemeStyles/utils.js#L631

To:

```js
str = str.replace(new RegExp(`"${alias.prev}":`, 'gi'), `"${alias.curr}":`);
```

To make it easier, you can change this directly on the distributable to test, located on `node_modules/@lightningjs/ui-components/dist/lightning-ui-components.min.mjs`:

From:

```js
n=n.replaceAll('"'.concat(t.prev,'":'),'"'.concat(t.curr,'":'))
```

To:

```js
n=n.replace(new RegExp('"'.concat(t.prev,'":'), 'gi'),'"'.concat(t.curr,'":'))
```

**Comment by Laerso-nos** [comment]
Hello @ricardo-a-alves-alb 
Thank you for your support.

With your suggested change **it is working fine again.**

So now, should we wait a new release on your side to bump the ui-components version on our project?

**Comment by ricardo-a-alves-alb** [comment]
> Hello @ricardo-a-alves-alb Thank you for your support.
> 
> With your suggested change **it is working fine again.**
> 
> So now, should we wait a new release on your side to bump the ui-components version on our project?

We should wait for the UI-Components team to add a polyfill for `replaceAll` and release a new version.

PR #386 and/or #387 addresses this.

**Comment by Laerso-nos** [comment]
@ricardo-a-alves-alb Sure, I'll keep an eye on those. Thank you one more time

**Comment by Laerso-nos** [comment]
Solved with release `2.14.3`
Closing the issue.
